connect-inject/endpoints_controller_test.go

```diff
@@ -925,11 +927,321 @@ func TestReconcileUpdateEndpoint(t *testing.T) {
 		expectedProxySvcInstances  []*api.CatalogService
 		expectedAgentHealthChecks  []*api.AgentCheck
 	}{
+		// Legacy services are not managed by endpoints controller, but endpoints controller
+		// will still add/update the legacy service's health checks.
+		{
```

Comment:
To reviewers since EP controller tests are long:
The tests I added were that for a legacy service: 
- the health check is added when the pod is healthy
- the health check is added when the pod is unhealthy
- the health check is updated from healthy --> unhealthy
- the health check is updated from unhealthy --> healthy

Since when an agent is rolled it will lose the health checks, I wanted to test that they are added back for a legacy service, and that they are updated if they already exist.